vision_evaluation/evaluators.py

```diff
@@ -675,3 +675,84 @@ def _per_class_calc(self, predictions, targets):
 
     def _get_id(self):
         return f'mAP_{self.n_points}_points'
+
+
+class ImageCaptionEvaluatorBase(Evaluator):
+    def __init__(self, metric):
+        super(ImageCaptionEvaluatorBase, self).__init__()
+        self.predictions = []
+        self.targets = []
+        self.metric = metric
+
+    def add_predictions(self, predictions, targets):
```

Comment:
The method name "add_predictions" indicate that it is adding predictions instead of setting predictions.